### DISCUSSION TITLE
More global settings, please!

### DISCUSSION THREAD

**Discussion by katycroffbell** [knowledgefutures/pubpub#2093] [Ideas]
It would be awesome to have more global features that can be set at the community settings level, from colors to toggling figures & tables. When building a big pubpub, it's *really* irritating to have to set everything by hand for *every* pub and page.

**Comment by idreyn** [comment]
We are working on this right now, and we hope to launch a first version in the next month or two! To start, we'll allow you to set these Pub settings at the Community or Collection level:

- Pub header theme
- Citation style
- Figure labeling settings
- License and copyright
- Pub Connections display settings
- _(Taking suggestions for more)_

Pubs will receive these styles, but will be able to override them for further customization. Hopefully this will cover a lot of what you need to set over and over to maintain a consistent look and feel. Once we have the system in place we'll be able to add more of these "cascading" settings easily.

We haven't started thinking about Pages quite yet. I'd be interested to hear what you'd like to globally configure for those!

**Reply by kyledenlinger** [reply]
I've been thinking how nice it would be to set a global color scheme and preview image style/texture/pattern that can generate more dynamic preview images and headers without me having to find images for these things. Something like what's offered at https://bgjar.com/ or https://doodad.dev/pattern-generator/ or any number of other background/texture generators.

**Reply by tefkah** [reply]
Oo that would be really useful @denlinkd! Atm we also almost always choose the minimal or small display style for pubs as we don't have the resources to create them ourselves and the default preview image doesn't really fit the rest of our color scheme.
Those resources you shared are really useful, thanks!

**Comment by katycroffbell** [comment]
Fantastic! Setting colors for banners and headers could also be great (for both pubs & pages). And other style options for page headers (narrow vs tall, text justification, etc.)--having a global/cascading heading to apply to all, but also customization for different settings sounds great.

**Comment by katycroffbell** [comment]
Related, it would be great to be able to crop the photos that you upload for previews & headers within the pubpub settings interface. i have folders and folders of crops for landscape (banners) vs portrait (previews) and it would be great if that's just all done in the interface.

**Comment by kyledenlinger** [comment]
Wondering if there's been any movement on this recently. Asking because I'm about to do a BUNCH of settings changes that would be so much more easily done in a global settings manager. 😬 🤞

**Reply by gabestein** [reply]
Hey Kyle — yup, we're actually working on it now. Not sure exactly when it'll be live, but weeks, not months. I'm not 100% sure if the first iteration will allow you to override existing settings, but I can let you know next week — and if not, we can also help with that in the backend if it doesn't.
